engine/access/rpc/backend/backend_transactions.go

```diff
@@ -696,12 +694,9 @@ func (b *backendTransactions) getTransactionResultFromAnyExeNode(
 				Msg("Successfully got transaction results from any node")
 			return resp, nil
 		}
-		if status.Code(err) == codes.NotFound {
```

Comment:
previously, we were failing early if the TX wasn't found on any EN. This made sense when we only queried the ENs that had reported an `ExecutionResult` for the block. Now that we query any EN, this should be relaxed.